platforms/core-runtime/java-language-extensions/build.gradle.kts

```diff
@@ -3,7 +3,7 @@ plugins {
     id("gradlebuild.publish-public-libraries")
 }
 
-description = "Common shared annotations"
+description = "Extensions to the Java language that are used across the Gradle codebase"
```

Comment:
🤔 Do you consider the `functional` subproject to be something like `java-language-extensions-jdk8` or is it something different?

Reply:
That's a good question. I think it would be good to keep things fine-grained, and leave the `:functional` subproject to exist on its own. My concern is we might open the floodgates, and then all kinds of data structures will end up in here. We can certainly merge things later on if we see it as a better option after some use.

Another, related idea is that we could have a bunch of things available by default to certain Gradle subprojects. `:java-language-extensions` is one thing that should be a default API dependency everywhere. We could also expose `:functional` to anything Java 8-compatible without asking. And Guava should be there for anything that's not exposed to workers etc. This PR is does not address these default dependencies, but I think we should do so later on.